### PR TITLE
feat/AB#107501_EIS-user-activity-track-page-title

### PR DESCRIPTION
**Pull request by unai-reliefapp**
# Description

feat: add current page title or navigation section to the activity metadata object using the breadcrumbs value

## Useful links

- [Link to ticket](https://dev.azure.com/WHOHQ/EMSSAFE/_workitems/edit/107501)
- [Link to back-end branch if any](https://github.com/ReliefApplications/ems-backend/pull/1173)

## Type of change

Please delete options that are not relevant.

- [X] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce.

- [ ] Test A
- [ ] Test B

## Screenshots

Please include screenshots of this change. If this issue is only back-end related, and does not involve any visual change of the platform, you can skip this part.

# Checklist:

( * == Mandatory ) 

- [X] * I have set myself as assignee of the pull request
- [X] * My code follows the style guidelines of this project
- [X] * Linting does not generate new warnings
- [X] * I have performed a self-review of my own code
- [X] * I have put the ticket for review, adding the oort-frontend team to the list of reviewers
- [X] * I have commented my code, particularly in hard-to-understand areas
- [X] * I have put JSDoc comment in all required places
- [X] * My changes generate no new warnings
- [X] * I have included screenshots describing my changes if relevant
- [X] * I have selected labels in the Pull Request, according to the changes with code brings
- [ ] I have made corresponding changes to the documentation ( if required )
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

# More explanation
https://www.loom.com/share/05a716d61b9744faaf51fb304c21d1e5?sid=f87cf896-582a-4f76-93ae-8ceed801b145

### CODE REVIEW

**Comment by unai-reliefapp** [comment]
Hi @AntoineRelief , I'll add the display of the url and page title in the activity table now, I pushed the logic to save the linked title of the page using breadcrumbs feature.
If you can take a look and let me know if all the needed cases are covered
KR, Unai

**Comment by AntoineRelief** [comment]
@unai-reliefapp 
Nice approach, I'll give it a look

**Comment by unai-reliefapp** [comment]
Hi @AntoineRelief , display title and url like that?
image

**Comment by AntoineRelief** [comment]
@unai-reliefapp 
It's fine, not 100% convinced by colors & font sizes, but I'll play with it when reviewing, that would be too long to find the correct display with 4 hands

**Comment by unai-reliefapp** [comment]
@AntoineRelief  And for the xlsx i did not really understand how is supposed to be, i can push the backend with the title column set like that
image

**Comment by AntoineRelief** [comment]
@unai-reliefapp 
in the xlsx you remove the metadata column and you only keep the title

**Comment by AntoineRelief** [comment]
@unai-reliefapp 

it seems to work for application pages that we create as admins, but it doesn't work for pages that are defined by the system ( admin pages )

For example:
Capture d’écran 2024-12-11 à 10 10 16

Some pages don't generate any log ( application settings for example ), while others would generate duplicate logs, and you can see one of the two logs it generates is not valid

File: apps/back-office/src/app/application/application-routing.module.ts [hunk 1]
```diff
@@ -189,6 +194,11 @@ const routes: Routes = [
           import('./pages/workflow/workflow.module').then(
             (m) => m.WorkflowModule
           ),
+        data: {
+          breadcrumb: {
```
Comment (AntoineRelief) [inline apps/back-office/src/app/application/application-routing.module.ts]: @unai-reliefapp 
we also need to do similar changes to front-office & app-preview

**Comment by unai-reliefapp** [comment]
> @unai-reliefapp
> 
> it seems to work for application pages that we create as admins, but it doesn't work for pages that are defined by the system ( admin pages )
> 
> For example: Capture d’écran 2024-12-11 à 10 10 16
> 
> Some pages don't generate any log ( application settings for example ), while others would generate duplicate logs, and you can see one of the two logs it generates is not valid

Defined by the system you mean this pages?
image

Or is there any other to take in account

**Comment by AntoineRelief** [comment]
@unai-reliefapp 
yes, all these pages + the one that are outside of applications, in the back-office default view

**Comment by unai-reliefapp** [comment]
@AntoineRelief fixes pushed and all set for front and back office
front office dashboard on workflow does not work(i dont know why the is inside workflow check is not added in the dashboard for front-office, i can add it but i first i wanted to check with you)

I will finish the web-application section tomorrow first thing in the morning

KR, Unai

**Comment by unai-reliefapp** [comment]
@AntoineRelief  Just FYI, the applicationId saved on the activities for applications with shortcut, it's the shortcut. 
image
So the activities save fail in those cases because of this line in the backend activities /post method
`applicationId: new Types.ObjectId(body.metadata.applicationId),`

```
Argument passed in must be a string of 12 bytes or a string of 24 hex characters or an integer {"stack":"BSONError: Argument passed in must be a string of 12 bytes or a string of 24 hex characters or an integer\n    at new ObjectId
```

**Comment by AntoineRelief** [comment]
> @AntoineRelief Just FYI, the applicationId saved on the activities for applications with shortcut, it's the shortcut. image So the activities save fail in those cases because of this line in the backend activities /post method `applicationId: new Types.ObjectId(body.metadata.applicationId),`
> 
> ```
> Argument passed in must be a string of 12 bytes or a string of 24 hex characters or an integer {"stack":"BSONError: Argument passed in must be a string of 12 bytes or a string of 24 hex characters or an integer\n    at new ObjectId
> ```

hey @unai-reliefapp 

yep, noticed it yesterday as well
I deployed a first version on DEV, but the way we get the application id is not working at all in DEV, and locally, it's not working if we use an app shortcut
I'll have a look at that this morning

**Comment by unai-reliefapp** [comment]
@AntoineRelief All integrations pushed, let me know if there is anything to be changed

**Pull request merged**